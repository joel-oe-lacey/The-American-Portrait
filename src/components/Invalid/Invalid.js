import React from 'react';
import './Invalid.scss';
import { Link } from 'react-router-dom';

const Invalid = () => {
    return (
        <section className="error">
            <section className="error-center">
                <h1>
                    Page Not Found, navigate back to the timeline below:
                </h1>
                <Link to="/timeline">Timeline</Link>
            </section>
        </section>
    )
}

export default Invalid;