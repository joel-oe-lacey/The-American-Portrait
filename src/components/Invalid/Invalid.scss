.error {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.error-center {
    height: 40%;
    width: 40%;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    text-align: center;

    h1 {
        font-size: 2em;
    }

    h2 {
        font-size: 1.5em;
    }
}