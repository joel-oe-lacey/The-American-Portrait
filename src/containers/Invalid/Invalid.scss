@import '../../variables';

.error {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background-color: #202020;
    color: white;
}

.error-center {
    height: 40%;
    width: 40%;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    text-align: center;

    h1 {
        font-size: 2em;
    }

    h2 {
        font-size: 1.5em;
    }
}

.home-link {
    text-decoration: none;
    font-size: 2em;
    font-weight: bold;
    color: white;
    border: 1px solid white;
    width: 50%;
    align-self: center;
    @include roundedBorder(35%);
}