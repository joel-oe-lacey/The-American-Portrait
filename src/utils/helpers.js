export class ArtPiece {
    primaryimageurl;
    title;
    objectid;
    description;
    century;
    dated;
    culture;
    classification;
    technique;
    medium;
    colors;
    creditline;
    labeltext;

    constructor(rawArtData) {
        Object.assign(this, rawArtData);
    }
};

export const restructureArtPiece = (rawArtData) => {
    const {
        title,
        titles,
        objectid,
        description,
        provenance,
        commentary,
        labeltext,
        classification,
        creditline,
        century,
        culture,
        medium,
        videos,
        datebegin,
        dateend,
        dated,
        period,
        technique,
        colors,
        primaryimageurl,
        images,
    } = rawArtData;

    return {
        title,
        titles,
        objectid,
        description,
        provenance,
        commentary,
        labeltext,
        classification,
        creditline,
        century,
        culture,
        medium,
        videos,
        datebegin,
        dateend,
        dated,
        period,
        technique,
        colors,
        primaryimageurl,
        images
    }
}

export const bucketArtByDate = (allArt) => {
    const recompileCollection = (bucketedCollection) => {
        return Object.keys(bucketedCollection).reduce((flatColl, dateKey) => {
            flatColl = flatColl.concat(bucketedCollection[dateKey])
            return flatColl;
        }, [])
    }

    const bucketedCollection = allArt.reduce((bucketedArt, piece) => {
        const date = piece.dateend;
        switch(true) {
            case (date < 1800): 
                if (bucketedArt.toEighteenHundreds.length < 10) {
                    bucketedArt.toEighteenHundreds.push(piece)
                }
                break;
            case (date <= 1850):
                if (bucketedArt.eighteenFiftys.length < 10) {
                    bucketedArt.eighteenFiftys.push(piece)
                }
                break;
            case (date <= 1880):
                if (bucketedArt.eighteenEighty.length < 10) {
                    bucketedArt.eighteenEighty.push(piece)
                }
                break;
            case (date <= 1900):
                if (bucketedArt.toNineteenHundreds.length < 10) {
                    bucketedArt.toNineteenHundreds.push(piece)
                }
                break;
            case (date <= 1920):
                if (bucketedArt.toNineteenTwenties.length < 10) {
                    bucketedArt.toNineteenTwenties.push(piece)
                }
                break;
            case (date <= 1940):
                if (bucketedArt.toNineteenFourties.length < 10) {
                    bucketedArt.toNineteenFourties.push(piece)
                }
                break;
            case (date <= 1960):
                if (bucketedArt.toNineteenSixties.length < 10) {
                    bucketedArt.toNineteenSixties.push(piece)
                }
                break;
            case (date <= 1980):
                if (bucketedArt.toNineteenEighties.length < 10) {
                    bucketedArt.toNineteenEighties.push(piece)
                }
                break;
            case (date <= 2000):
                if (bucketedArt.toTwoThousands.length < 10) {
                    bucketedArt.toTwoThousands.push(piece)
                }
                break;
            case (date <= 2020):
                if (bucketedArt.toTwentyTwenties.length < 10) {
                    bucketedArt.toTwentyTwenties.push(piece)
                }
                break;
            default:
                break;
        }
        return bucketedArt;
    }, {
        "toEighteenHundreds": [],
        "toEighteenFiftys": [],
        "toEighteenEighty": [],
        "toNineteenHundreds": [],
        "toNineteenTwenties": [],
        "toNineteenFourties": [],
        "toNineteenSixties": [],
        "toNineteenEighties": [],
        "toTwoThousands": [],
        "toTwentyTwenties": [],
    })

    return recompileCollection(bucketedCollection);
}

