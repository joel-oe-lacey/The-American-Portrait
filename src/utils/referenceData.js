export const introText = ['People are trapped in history, and history is trapped in them" said James Baldwin in his 1953 essay "Stranger in the Village". At the time he spoke of the indivisibility of black culture from the larger collective American ethos and culture.', 
'A continuation of that allegory, I believe, also speaks to the importance of remaining in touch with our collective history. It, in similar duality, is a part of us all.', 'The American Portrait looks to reunite us with a visual piece of that history. Utilizing the Harvard Museum API, you can view a timeline of historic photos by state, ten per decade where available.','While incomplete and historically biased, these photos offer us at very least a partial window into some of the lives, events, and adolescent places that would go on to form the nation as we know it. ']

export const stateNames = ['Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California', 'Colorado', 'Connecticut', 'Delaware', 'Florida', 'Georgia', 'Hawaii', 'Idaho', 'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi', 'Missouri', 'Montana', 'Nebraska', 'Nevada', 'New Hampshire', 'New Jersey', 'New Mexico', 'New York', 'North Carolina', 'North Dakota', 'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania', 'Puerto Rico', 'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee', 'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington', 'West Virginia', 'Wisconsin', 'Wyoming']

export const mockCollectionPrimary = {
    "title": "House and [blue] Car, Near Akron, Alabama",
    "objectid": 354191,
    "creditline": "Harvard Art Museums/Fogg Museum, Fund for the Acquisition of Photographs",
    "century": "20th century",
    "technique": "Dye transfer print",
    "dateend": 1978,
    "dated": "1978, printed 1991",
    "people": [
        {
        "alphasort": "Christenberry, William",
        "birthplace": "Tascaloosa, Alabama",
        "name": "William Christenberry",
        "prefix": null,
        "personid": 62113,
        "gender": "unknown",
        "role": "Artist",
        "displayorder": 1,
        "culture": "American",
        "displaydate": "born 1936",
        "deathplace": null,
        "displayname": "William Christenberry"
        }
    ],
    "primaryimageurl": "https://nrs.harvard.edu/urn-3:HUAM:761794"
}

export const mockCollectionSecondary = {
    "title": "Night Spot, Marion, Alabama",
    "objectid": 354192,
    "creditline": "Harvard Art Museums/Fogg Museum, Fund for the Acquisition of Photographs",
    "century": "20th century",
    "technique": "Dye transfer print",
    "dateend": 1972,
    "dated": "1972, printed 1991",
    "people": [
        {
            "alphasort": "Christenberry, William",
            "birthplace": "Tascaloosa, Alabama",
            "name": "William Christenberry",
            "prefix": null,
            "personid": 62113,
            "gender": "unknown",
            "role": "Artist",
            "displayorder": 1,
            "culture": "American",
            "displaydate": "born 1936",
            "deathplace": null,
            "displayname": "William Christenberry"
        }
    ],
    "primaryimageurl": "https://nrs.harvard.edu/urn-3:HUAM:761793"
}