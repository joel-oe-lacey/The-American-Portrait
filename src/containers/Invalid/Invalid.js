import React from 'react';
import './Invalid.scss';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export const Invalid = ({ collections }) => {
    // const msg = (collections.length ? 'Page Not Found.' : 'Insufficient historical record, please try another state.')

    return (
        <section className="error">
            <section className="error-center">
                <h1>
                    Page error, or there are no available photos.
                </h1>
                <h2>
                    Please begin again.
                </h2>
                <Link to="/" className="home-link">Home</Link>
            </section>
        </section>
    )
}

export const mapStateToProps = state => ({
    collections: state.collections,
});

export default connect(mapStateToProps)(Invalid)