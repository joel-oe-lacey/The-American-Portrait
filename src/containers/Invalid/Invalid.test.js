import React from 'react';
import { Invalid } from './Invalid';
import { shallow } from 'enzyme';
import { mockCollectionPrimary, mockCollectionSecondary } from '../../utils/referenceData';

describe('Invalid Path', () => {
    it('should match a snapshot as expected', () => {
        const wrapper = shallow(<Invalid collections={[mockCollectionPrimary, mockCollectionSecondary]} />)

        expect(wrapper).toMatchSnapshot()
    })
})
