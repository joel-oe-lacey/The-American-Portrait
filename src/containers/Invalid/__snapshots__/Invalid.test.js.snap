// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid Path should match a snapshot as expected 1`] = `
<section
  className="error"
>
  <section
    className="error-center"
  >
    <h1>
      Page Not Found.
    </h1>
    <Link
      to="/"
    >
      Home
    </Link>
  </section>
</section>
`;
